### ISSUE TITLE
Weird behavior (probably numerics) in iaf_psc_exp_ps for tau_m=tau_syn

### ISSUE THREAD

**Issue by healther** [nest/nest-simulator#1087]
It seems as if the `iaf_psc_exp_ps` model has problems integrating synaptic input for `tau_m=tau_syn_*`. We think it is a problem with the implementation of the alpha-function, as it works for even slight differences between `tau_m` and `tau_syn`.

The following code produces the 4 attached plots (executed with nest 2.14.0 and reproduced by @jakobj on current master), bug found and reproducer code by @JulianGoeltz
```
import matplotlib.pyplot as plt
import nest
def doit(ratio, model):
    print("MWE_plot_{}_{}.svg".format(model, ratio))
    tau_syn = 10.
    nest.ResetKernel()
    nest.ResetNetwork()
    x = nest.Create(model, 1, {'tau_m': ratio * tau_syn, 'tau_syn_ex': tau_syn, 'tau_syn_in': tau_syn,
                                       'V_th': -10., 'E_L': -20., 'V_m': -15.})
    y = nest.Create('spike_generator', 1, {'spike_times': [53., 55., 57.]})
    nest.Connect(y, x, syn_spec={'weight': 240.})
    vm = nest.Create('voltmeter')
    nest.Connect(vm, x)
    nest.Simulate(100.)
    ev = nest.GetStatus(vm, 'events')[0]
    fig = plt.figure()
    plt.plot(ev['times'], ev['V_m'])
    fig.savefig("MWE_plot_{}_{}.svg".format(model, ratio))

for model in ['iaf_psc_exp', 'iaf_psc_exp_ps']:
    for ratio in [1., 2.]:
        doit(ratio, model)
```

[MWE_plot_iaf_psc_exp_1.0.pdf](https://github.com/nest/nest-simulator/files/2664051/MWE_plot_iaf_psc_exp_1.0.pdf)
[MWE_plot_iaf_psc_exp_2.0.pdf](https://github.com/nest/nest-simulator/files/2664052/MWE_plot_iaf_psc_exp_2.0.pdf)
[MWE_plot_iaf_psc_exp_ps_1.0.pdf](https://github.com/nest/nest-simulator/files/2664053/MWE_plot_iaf_psc_exp_ps_1.0.pdf)
[MWE_plot_iaf_psc_exp_ps_2.0.pdf](https://github.com/nest/nest-simulator/files/2664054/MWE_plot_iaf_psc_exp_ps_2.0.pdf)

**Comment by jakobj** [comment]
i've created the following patch that seems to eliminate the problem at least in this test case:

```diff
diff --git a/precise/iaf_psc_exp_ps.cpp b/precise/iaf_psc_exp_ps.cpp
index ba2a1a16..3eca1b89 100644
--- a/precise/iaf_psc_exp_ps.cpp
+++ b/precise/iaf_psc_exp_ps.cpp
@@ -497,10 +497,9 @@ nest::iaf_psc_exp_ps::propagate_( const double dt )
     const double expm1_tau_m = numerics::expm1( -dt / P_.tau_m_ );
 
     const double P20 = -P_.tau_m_ / P_.c_m_ * expm1_tau_m;
-    const double P21_ex = -P_.tau_m_ * P_.tau_ex_ / ( P_.tau_m_ - P_.tau_ex_ )
-      / P_.c_m_ * ( expm1_tau_ex - expm1_tau_m );
-    const double P21_in = -P_.tau_m_ * P_.tau_in_ / ( P_.tau_m_ - P_.tau_in_ )
-      / P_.c_m_ * ( expm1_tau_in - expm1_tau_m );
+
+    const double P21_ex = propagator_32( P_.tau_ex_, P_.tau_m_, P_.c_m_, dt );
+    const double P21_in = propagator_32( P_.tau_in_, P_.tau_m_, P_.c_m_, dt );
 
     S_.y2_ = P20 * ( P_.I_e_ + S_.y0_ ) + P21_ex * S_.y1_ex_
       + P21_in * S_.y1_in_ + expm1_tau_m * S_.y2_ + S_.y2_;
@@ -583,15 +582,12 @@ nest::iaf_psc_exp_ps::bisectioning_( const double dt ) const
 
     div *= 2.0;
 
-    const double expm1_tau_ex = numerics::expm1( -root / P_.tau_ex_ );
-    const double expm1_tau_in = numerics::expm1( -root / P_.tau_in_ );
     const double expm1_tau_m = numerics::expm1( -root / P_.tau_m_ );
 
     const double P20 = -P_.tau_m_ / P_.c_m_ * expm1_tau_m;
-    const double P21_ex = -P_.tau_m_ * P_.tau_ex_ / ( P_.tau_m_ - P_.tau_ex_ )
-      / P_.c_m_ * ( expm1_tau_ex - expm1_tau_m );
-    const double P21_in = -P_.tau_m_ * P_.tau_in_ / ( P_.tau_m_ - P_.tau_in_ )
-      / P_.c_m_ * ( expm1_tau_in - expm1_tau_m );
+
+    const double P21_ex = propagator_32( P_.tau_ex_, P_.tau_m_, P_.c_m_, root );
+    const double P21_in = propagator_32( P_.tau_in_, P_.tau_m_, P_.c_m_, root );
 
     y2_root = P20 * ( P_.I_e_ + V_.y0_before_ ) + P21_ex * V_.y1_ex_before_
       + P21_in * V_.y1_in_before_ + expm1_tau_m * V_.y2_before_ + V_.y2_before_;
```

@JulianGoeltz please let us know whether this works also in more general cases
@jschuecker i think you've fixed a similar issue like this before, maybe you could have a look?

**Comment by healther** [comment]
FYI: Also this code executed with the unpatched nest@2.14.0 takes very long to terminate (i.e. we didn't wait until it finished, but >2mins) for the precise model, the imprecise one does terminate, but produces a valid looking plot for the normal model (I wouldn't bet on it not being numerically unstable though).

With the path by @jakobj both go through without any problems
```
import nest
def doit(ratio, model):
    print("MWE_plot_{}_{}.svg".format(model, ratio))
    tau_syn = 10.0
    nest.ResetKernel()
    nest.ResetNetwork()
    x = nest.Create(model, 1, {'tau_m': ratio * tau_syn, 'tau_syn_ex': tau_syn, 'tau_syn_in': tau_syn,
                                       'V_th': -10., 'E_L': -20., 'V_m': -15.})
    y = nest.Create('spike_generator', 1, {'spike_times': [53., 55., 57.]})
    nest.Connect(y, x, syn_spec={'weight': 240.})
    vm = nest.Create('voltmeter')
    nest.Connect(vm, x)
    nest.Simulate(100.)
    ev = nest.GetStatus(vm, 'events')[0]
    fig = plt.figure()
    plt.plot(ev['times'], ev['V_m'])
    plt.show()
    fig.savefig("MWE_plot_{}_{}.svg".format(model, ratio))

for model in ['iaf_psc_exp', 'iaf_psc_exp_ps']:
    for ratio in [1.000001]:
        doit(ratio, model)
print(nest.__file__)
```

**Comment by heplesser** [comment]
@jakobj I think you were referring to #71. Could you create a PR?

**Comment by jakobj** [comment]
sure, i'll create a PR

**Comment by heplesser** [comment]
Fixed by #1112, thus closing.
